enterprise/server/tasksize/tasksize.go

```diff
@@ -412,33 +439,25 @@ func Estimate(task *repb.ExecutionTask) *scpb.TaskSize {
 		}
 	}
 
-	if props.EstimatedComputeUnits > 0 {
-		cpuEstimate = int64(props.EstimatedComputeUnits * ComputeUnitsToMilliCPU)
-		memEstimate = int64(props.EstimatedComputeUnits * ComputeUnitsToRAMBytes)
-	}
-	if props.EstimatedMilliCPU > 0 {
-		cpuEstimate = props.EstimatedMilliCPU
-	}
-	if props.EstimatedMemoryBytes > 0 {
-		memEstimate = props.EstimatedMemoryBytes
-	}
-	if props.EstimatedFreeDiskBytes > 0 {
-		freeDiskEstimate = props.EstimatedFreeDiskBytes
-	}
-	if freeDiskEstimate > MaxEstimatedFreeDisk {
-		log.Infof("Task %q requested %d free disk which is more than the max %d", task.GetExecutionId(), freeDiskEstimate, MaxEstimatedFreeDisk)
-		freeDiskEstimate = MaxEstimatedFreeDisk
-	}
-
-	return applyMinimums(task, &scpb.TaskSize{
+	return applyLimits(task, &scpb.TaskSize{
 		EstimatedMemoryBytes:   memEstimate,
 		EstimatedMilliCpu:      cpuEstimate,
 		EstimatedFreeDiskBytes: freeDiskEstimate,
-		CustomResources:        props.CustomResources,
 	})
 }
 
-func applyMinimums(task *repb.ExecutionTask, size *scpb.TaskSize) *scpb.TaskSize {
+// Combine joins two task sizes, generally by taking the max of each field. It
+// clamps values to allowed ranges.
+func Combine(task *repb.ExecutionTask, a, b *scpb.TaskSize) *scpb.TaskSize {
```

Comment:
nit: do you think there will be use cases for this outside of merging the default+requested sizes? if not, it might make sense to explicitly call Max() and ApplyLimits() as 2 separate functions, instead of having this Combine() function, since it feels a bit niche